### PR TITLE
ICS20 authz E2E happy path

### PR DESCRIPTION
**Pull request by chatton**
## Description

ref: #2964

### Commit Message / Changelog Entry

```bash
e2e: added happy path e2e test for ICS20 authz integration
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -539,7 +542,32 @@ func (s *E2ETestSuite) QueryModuleAccountAddress(ctx context.Context, moduleName
...
+
+// getValidatorsAndFullNodes returns the number of validators and full nodes respectively that should be used for
+// the test. If the test is running in CI, more nodes are used, when running locally a single node is used to
+// use less resources and allow the tests to run faster.
+func getValidatorsAndFullNodes() (int, int) {
+	if testconfig.IsCI() {
+		return 4, 1
+	}
+	return 1, 0
+}
```
Comment (chatton) [inline e2e/testsuite/testsuite.go:573]: unrelated QOL change to allow running tests locally to run faster.

Comment (damiannolan) [reply e2e/testsuite/testsuite.go:573]: nice

File: e2e/tests/transfer/authz_test.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+				Expiration: nil,
+			},
+		}
+
+		resp, err := suite.BroadcastMessages(context.TODO(), chainA, granterWallet, msgGrant)
+		suite.AssertValidTxResponse(resp)
+		suite.Require().NoError(err)
+	})
+
+	t.Run("exec MsgTransfer", func(t *testing.T) {
```
Comment (damiannolan) [inline e2e/tests/transfer/authz_test.go:78]: wonder if we should reword this a little 

```suggestion
	t.Run("broadcast MsgExec for ibc MsgTransfer", func(t *testing.T) {
```

**Review by damiannolan — APPROVED** [review]
Nice work! LGTM 🚀

File: e2e/testconfig/testconfig.go [hunk 3]
```diff
@@ -129,6 +130,13 @@ func GetChainBTag() string {
 	return chainBTag
 }
 
+// IsCI returns true if the tests are running in CI, false is returned
+// if the tests are running locally.
+// Note: github actions passes a CI env value of true by default to all runners.
```
Comment (charleenfei) [inline e2e/testconfig/testconfig.go:135]: nice 🎉

**Review by charleenfei — APPROVED** [review]
nice work :)

File: e2e/testsuite/testsuite.go (hunk 1 above)
Comment (charleenfei) [reply e2e/testsuite/testsuite.go:573]: yes i love this!
